step-templates/aws-configure-lambda-alias.json

```diff
@@ -0,0 +1,89 @@
Highlight \"The function's alias $functionAliasName does not exist.\"\n      Write-Host \"If you see an error right here you can safely ignore that.\"\n\t  $aliasInformation = $null\n  }   \n  else\n  {\n\t  Write-Highlight \"The function's alias $functionAliasName already exists.\"\n\t  $aliasInformation = $aliasInformation | ConvertFrom-JSON\n\t  Write-Host $aliasInformation\n  }  \n}\ncatch\n{\n  Write-Host \"The alias specified $functionAliasName does not exist for $functionName.  Will create a new alias with that name.\"\n  $aliasInformation = $null\n}\n\nif ($null -ne $aliasInformation)\n{\n  \tWrite-Host \"Comparing the existing alias version $($aliasInformation.FunctionVersion) with the the published version $versionToUpdateTo\"\n                    \n   \tif ($aliasInformation.FunctionVersion -ne $versionToUpdateTo)\n    {\n    \tWrite-Host \"The alias $functionAliasName version $($aliasInformation.FunctionVersion) does not equal the published version $versionToUpdateTo\"\n            \n        if ($newVersionPercent -eq 100)\n        {\n         \tWrite-Highlight \"The percent for the new version of the function is 100%, updating the alias $functionAliasName to function version $versionToUpdateTo\"\n           \t$newAliasInformation = aws lambda update-alias --function-name \"$functionName\" --name \"$functionAliasName\" --function-version \"$versionToUpdateTo\" --routing-config \"AdditionalVersionWeights={}\"\n        }\n        else\n        {              \t\n            $newVersionPercent = $newVersionPercent / [double]100                       \n                \n            Write-Highlight \"Updating the alias $functionAliasName so $functionAliasPercent of all traffic is routed to $versionToUpdateTo\"\n\t\t\t  \n   \t        $newAliasInformation = aws lambda update-alias --function-name \"$functionName\" --name \"$functionAliasName\" --routing-config \"AdditionalVersionWeights={\"\"$versionToUpdateTo\"\"=$newVersionPercent}\"\n        }           \n    }\n    elseif ($newVersionPercent -eq 100)\n    {\n    \tWrite-Highlight \"The alias $functionAliasName is already pointed to $versionToUpdateTo and the percent sent in is 100, updating the function so all traffic is routed to that version.\"\n        $newAliasInformation = aws lambda update-alias --function-name \"$functionName\" --name \"$functionAliasName\" --routing-config \"AdditionalVersionWeights={}\"\n    }\n    else\n    {\n   \t\tWrite-Highlight \"The alias $functionAliasName is already pointed to $versionToUpdateTo.  Leaving as is.\"\n    }\n}\nelse\n{\n   \tWrite-Highlight \"Creating the alias $functionAliasName with the version $($publishedVersion.Version)\"\n \t$newAliasInformation = aws lambda create-alias --function-name \"$functionName\" --name \"$functionAliasName\" --function-version \"$versionToUpdateTo\"\n}\n\nif ($null -ne $newAliasInformation)\n{\n\tWrite-Host ($newAliasInformation | ConvertFrom-JSON)\n}\n\nWrite-Highlight \"The alias has finished updating.\""
```

Comment:
Line 43-44: is this just an implicit check to make sure the function exists and if it bombs, it bombs?
Line 126: $publishedVersion.Version Not sure where this comes from? Doesn't seem like it's set anywhere.

Reply:
Updated.

Reply:
May want to have some guard if the $functionName is typed wrong or doesn't exist? I wasn't sure if that's what your previous lines 43-44 were doing. Otherwise, not sure what `aws lambda list-versions-by-function` returns if functionName isn't found. Might be nice to call out if it's not found.

Reply:
It'll be fine, subsequent calls will fail with a message it cannot find the function.  I was doing that in case subsequent calls needed the function ARN, which it does not.